Backend/nix/services.nix

```diff
@@ -0,0 +1,128 @@
+          # See here about `pg_basebackup`: https://www.postgresql.org/docs/current/app-pgbasebackup.html
+          pg-backup-script = master-db-name: replica-db-name: port: pkgs.writeShellApplication {
+            name = "start-pg-basebackup";
+            runtimeInputs = with pkgs; [ config.services.postgres."${master-db-name}".package coreutils ];
+            text = ''
+              MASTER_DB_ABS_PATH=$(readlink -f ${config.services.postgres."${master-db-name}".dataDir})
+              REPLICA_DB_ABS_PATH=$(readlink -f ${config.services.postgres."${replica-db-name}".dataDir})
+              pg_basebackup -h "$MASTER_DB_ABS_PATH" -U repl_user --checkpoint=fast -D "$REPLICA_DB_ABS_PATH"  -R --slot=some_name  -C --port=${builtins.toString port}
+            '';
+          };
+        in
+        {
+          imports = [
+            inputs.services-flake.processComposeModules.default
+            inputs.passetto.processComposeModules.default
+          ];
+
+          services.postgres.db-primary = {
+            extensions = extensions: [
+              extensions.postgis
+            ];
+            initialDumps = [
+              ../dev/sql-seed/pre-init.sql
+              ../dev/sql-seed/rider-app-seed.sql
+              ../dev/local-testing-data/rider-app.sql
+              ../dev/sql-seed/public-transport-rider-platform-seed.sql
+              ../dev/local-testing-data/public-transport-rider-platform.sql
+              ../dev/sql-seed/mock-registry-seed.sql
+              ../dev/local-testing-data/mock-registry.sql
+              ../dev/sql-seed/scheduler-example-seed.sql
+              ../dev/sql-seed/dynamic-offer-driver-app-seed.sql
+              ../dev/local-testing-data/dynamic-offer-driver-app.sql
+              ../dev/sql-seed/rider-dashboard-seed.sql
+              ../dev/local-testing-data/rider-dashboard.sql
+              ../dev/sql-seed/provider-dashboard-seed.sql
+              ../dev/local-testing-data/provider-dashboard.sql
+              ../dev/sql-seed/special-zone-seed.sql
+              ../dev/local-testing-data/special-zone.sql
+            ];
+            port = 5432;
+            enable = true;
+            listen_addresses = "*";
+            hbaConf = [
+              { type = "host"; database = "all"; user = "repl_user"; address = "127.0.0.1/32"; method = "trust"; }
+            ];
+            initialScript.before = ''
+              CREATE USER atlas WITH PASSWORD 'atlas';
+              
+              CREATE USER repl_user replication;
+            '';
+          };
+
+          settings.processes.pg-basebackup-primary-db = {
+            command = pg-backup-script "db-primary" "db-replica" 5432;
+            depends_on."db-primary".condition = "process_healthy";
+          };
+
+          services.postgres.db-replica = {
+            depends_on."pg-base-backup-primary-db".condition = "process_completed_successfully";
```

Comment:
@shivaraj-bh There's a typo `pg-base-backup-primary-db` should be `pg-basebackup-primary-db`  as defined at line 60, because of this typo db-replica isn't starting.